src/Compilers/Core/Portable/InternalUtilities/FatalError.cs

```diff
@@ -11,17 +13,17 @@ namespace Microsoft.CodeAnalysis.ErrorReporting
 {
     internal static class FatalError
     {
-        private static Action<Exception> s_fatalHandler;
-        private static Action<Exception> s_nonFatalHandler;
+        private static Action<Exception>? s_fatalHandler;
+        private static Action<Exception>? s_nonFatalHandler;
 
-        private static Exception s_reportedException;
-        private static string s_reportedExceptionMessage;
+        private static Exception? s_reportedException;
+        private static string? s_reportedExceptionMessage;
 
         /// <summary>
         /// Set by the host to a fail fast trigger, 
         /// if the host desires to crash the process on a fatal exception.
         /// </summary>
-        public static Action<Exception> Handler
+        public static Action<Exception>? Handler
```

Comment:
Should these have a `DisallowNull` on them?

Reply:
The full preconditions for this method can't be represented by the available annotations. The use of `[DisallowNull]` wouldn't cause incorrect usage to be revealed by the compiler, so it seems reasonable to leave it out.

Reply:
On second thought, it would reveal someone incorrectly trying to _reset_ this property, when they should be calling `OverwriteHandler` for that instead.

Reply:
➡️ Fixed